tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+
+  return copy_file_and_signature (source_ref_file,
+				  source_ref_file_sig,
+				  target_ref_file,
+				  target_ref_file_sig,
+				  error);
+}
+
+static gboolean
+copy_extensions (GFile *source_repo,
+		 GFile *client_root,
+		 const gchar *ref,
+		 GError **error)
+{
+  g_autoptr(GFile) sysroot = get_sysroot_for_client (client_root);
+  g_autoptr(GFile) repo = get_repo_for_sysroot (sysroot);
+
+  if (!copy_branch_file (source_repo, repo, error))
+    return FALSE;
+
+  if (!copy_ref_file (source_repo, repo, ref, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+static const gchar *
+download_order_to_string (DownloadOrder order)
+{
+  switch (order)
+    {
+    case DOWNLOAD_MAIN:
+      return "main";
+
+    case DOWNLOAD_LAN:
+      return "lan";
+
+    case DOWNLOAD_MAIN_LAN:
+      return "main,lan";
+
+    case DOWNLOAD_LAN_MAIN:
+      return "lan,main";
+    }
+
+  g_assert_not_reached ();
+}
+
+static GFile *
+get_updater_dir_for_client (GFile *client_root)
+{
+  return g_file_get_child (client_root, "updater");
+}
+
+static GKeyFile *
+get_updater_config (DownloadOrder download_order)
+{
+  g_autoptr(GKeyFile) config = g_key_file_new ();
+
+  g_key_file_set_string (config, "Download", "Order", download_order_to_string (download_order));
+
+  return g_steal_pointer (&config);
+}
+
+static GKeyFile *
+get_hw_config (const gchar *vendor,
+               const gchar *product)
+{
+  g_autoptr(GKeyFile) hw = g_key_file_new ();
+
+  g_key_file_set_string (hw, "descriptors", "sys_vendor", vendor);
+  g_key_file_set_string (hw, "descriptors", "product_name", product);
+
+  return g_steal_pointer (&hw);
+}
+
+static GFile *
+updater_avahi_services_dir (GFile *updater_dir)
+{
+  return g_file_get_child (updater_dir, "avahi-services");
+}
+
+static GFile *
+updater_avahi_emulator_definitions_dir (GFile *updater_dir)
+{
+  return g_file_get_child (updater_dir, "avahi-emulator-definitions");
+}
+
+static GFile *
+updater_quit_file (GFile *updater_dir)
+{
+  return g_file_get_child (updater_dir, "quit-file");
+}
+
+static GFile *
+updater_config_file (GFile *updater_dir)
+{
+  return g_file_get_child (updater_dir, "config");
+}
+
+static GFile *
+updater_hw_file (GFile *updater_dir)
+{
+  return g_file_get_child (updater_dir, "hw");
+}
+
+static gboolean
+prepare_updater_dir (GFile *updater_dir,
+                     GKeyFile *config_file,
+                     GKeyFile *hw_file,
+                     GError **error)
+{
+  OSDEF;
+
+  if (!create_directory (OS (updater_avahi_services_dir (updater_dir)), error))
+    return FALSE;
+
+  if (!create_directory (OS (updater_avahi_emulator_definitions_dir (updater_dir)), error))
+    return FALSE;
+
+  if (!create_file (OS (updater_quit_file (updater_dir)), NULL, error))
+    return FALSE;
+
+  if (!save_key_file (OS (updater_config_file (updater_dir)), config_file, error))
+    return FALSE;
+
+  if (!save_key_file (OS (updater_hw_file (updater_dir)), hw_file, error))
```

Comment:
Not sure if they would help at all - the function take no reference to passed GFile, so they wouldn't sink them. Anyway, I rewrote it, so `OSDEF` is not more.